engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala

```diff
@@ -92,33 +92,40 @@ class EnsureCompiledJob(protected val files: Iterable[File])
       ctx.executionService.getContext.getTopScope.getModules.asScala
     ctx.executionService.getLogger
       .finest(s"Modules in scope: ${modulesInScope.map(_.getName)}")
-    val updates = modulesInScope.flatMap { module =>
+    modulesInScope.foreach { module =>
       compile(module)
       analyzeModuleInScope(module)
     }
-    sendIndexUpdateNotification(
-      Api.SuggestionsDatabaseIndexUpdateNotification(updates)
-    )
   }
 
   private def analyzeImport(
     module: Module
   )(implicit ctx: RuntimeContext): Unit = {
-    if (!module.isIndexed && module.getLiteralSource != null) {
+    if (
+      !module.isIndexed &&
+      module.getLiteralSource != null &&
+      module.getPath != null
```

Comment:
Not sure, but I think it is possible for the IDE to create an unsaved module? So we shouldn't really depend on the module being on-disk.

Reply:
Good catch. I added this case to the #1162 specification, which I'll do next.